run_test.sh

```diff
@@ -434,13 +441,82 @@ dkms_test.ko${mod_compression_ext}:
    - Use the dkms install command to reinstall any previous module version.
 Deleting module dkms_test-1.0 completely from the DKMS tree.
 EOF
-run_with_expected_output dkms_status_grep_dkms_test << EOF
+run_with_expected_output dkms_status_grep_dkms_module 'dkms_test' << EOF
 EOF
 
 echo 'Removing /usr/src/dkms_test-1.0'
 rm -r /usr/src/dkms_test-1.0
 
-echo 'Checking that the environment is clean'
+echo 'Checking that the environment is clean again'
+clean_dkms_env
+
+echo 'Running autoinstall error testing'
+
+echo 'Adding failing test module by directory'
+run_with_expected_output dkms add test/dkms_failing_test-1.0 << EOF
+Creating symlink /var/lib/dkms/dkms_failing_test/1.0/source -> /usr/src/dkms_failing_test-1.0
+EOF
+echo 'Running autoinstall with failing test module (expected error)'
+run_with_expected_error 11 dkms autoinstall << EOF
+Sign command: /usr/lib/linux-kbuild-5.10/scripts/sign-file
+Signing key: /var/lib/dkms/mok.key
+Public certificate (MOK): /var/lib/dkms/mok.pub
```

Comment:
The above 3 seem like hard-coded paths. I suspect that at least the `Sign command` will fail according to the distro - see how we sed those out in `run_with_expected_output`.

Perhaps we can do better and actually compare them, but for now that's good enough.

Reply:
I added a commit to pull the `sed` logic out into a helper and apply it to the `run_with_expected_error` function as well. That should at least make it easier to rip out or modify later.

Reply:
Agreed, thank you.